src/app/systelab-components/chip-button/chip-button.component.ts

```diff
@@ -64,9 +66,16 @@ export class ChipButtonComponent {
 	}
 
 	public addButtonITem() {
-		this.buttonList.push({name: this.i18nService.instant('COMMON_NEW'), id: this.buttonList.length + 1, isChecked: false});
-		this.selectItem(this.buttonList[this.buttonList.length - 1]);
-
+		let maxID = 0;
```

Comment:
I would refactor the following lines (from 69 to 73) to just the line below

`const maxID = Math.max(...this.buttonList.map(o => o.id), 0);`

Reply:
clever